src/plugin/any23/src/java/org/apache/nutch/any23/Any23ParseFilter.java

```diff
@@ -0,0 +1,165 @@
+   */
+  public final static String ANY23_TRIPLES = "Any23-Triples";
+
+  private static class Any23Parser {
+
+    Set<String> triples = null;
+
+    Any23Parser(String url, String htmlContent) throws TripleHandlerException {
+      triples = new TreeSet<String>();
+      try {
+        parse(url, htmlContent);
+      } catch (URISyntaxException e) {
+        throw new RuntimeException(e.getReason());
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+    }
+
+    /**
+     * Maintains a {@link java.util.Set} containing the triples
+     * @return a {@link java.util.Set} of triples.
+     */
+    public Set<String> getTriples() {
+      return triples;
+    }
+
+    private void parse(String url, String htmlContent) throws URISyntaxException, IOException, TripleHandlerException {
+      Any23 any23 = new Any23();
+      ByteArrayOutputStream baos = new ByteArrayOutputStream();
+      TripleHandler tHandler = new NTriplesWriter(baos);
+      BenchmarkTripleHandler bHandler = new BenchmarkTripleHandler(tHandler);
+      try {
+        any23.extract(htmlContent, url, "text/HTML", "UTF-8", bHandler);
+      } catch (Exception e) {
+        e.printStackTrace();
+      } finally {
+        tHandler.close();
+        bHandler.close();
+      }
+      //This merely prints out a report of the Any23 extraction.
+      LOG.info("Any23 report: " + bHandler.report());
+      String n3 = baos.toString("UTF-8");
+      // we split the triples stream by the occurrence of
+      // '\n' as this is a distinguishing feature of NTriples
+      // output serialization in Any23.
+      String[] triplesStrings = n3.split("\n");
+      Collections.addAll(triples, triplesStrings);
+    }
+  }
+
+  /**
+   * @see org.apache.hadoop.conf.Configurable#getConf()
+   */
+  @Override
+  public Configuration getConf() {
+    return this.conf;
+  }
+
+  /**
+   * @see org.apache.hadoop.conf.Configurable#setConf(org.apache.hadoop.conf.Configuration)
+   */
+  @Override
+  public void setConf(Configuration conf) {
+    this.conf = conf;
+  }
+
+  /**
+   * @see org.apache.nutch.parse.HtmlParseFilter#filter(Content, ParseResult, HTMLMetaTags, DocumentFragment)
+   */
+  @Override
+  public ParseResult filter(Content content, ParseResult parseResult, HTMLMetaTags metaTags, DocumentFragment doc) {
+
+    Any23Parser parser = null;
+    try {
+      String htmlContent = new String(content.getContent(), Charset.forName("UTF-8"));
+      parser = new Any23Parser(content.getUrl(), htmlContent);
+    } catch (TripleHandlerException e) {
+      throw new RuntimeException("Error running Any23 parser: " + e.getMessage());
+    }
+    Set<String> triples = parser.getTriples();
+    // can't store multiple values in page metadata -> separate by tabs
+    StringBuilder sb = new StringBuilder();
+
+    Parse parse = parseResult.get(content.getUrl());
+    Metadata metadata = parse.getData().getParseMeta();
+
+    for (String triple : triples) {
+      sb.append(triple);
```

Comment:
Previously we discussed and agreed that this was not an optional solution for associating triples with the Metadata. I still agree with that. We need to think of a more efficient manner for persisting triples.

Reply:
For the time being, I would move forward with implementing this. We can work on improving an appropriate storage mechanisms/location for Any23 extractions later on.

Reply:
Hi @lewismc taking over this PR, going to adapt the format of the triplets to an array of objects if that's ok.

(example data taken from http://dbpedia.org/data/Z%C3%BCrich.ntriples)

```
triples: [
  {
    key: 'http://www.w3.org/2002/07/owl#sameAs',
    short_key: 'sameAs', // for convenience
    value: http://rdf.freebase.com/ns/m.08966'',
  },
  {
    key: 'http://dbpedia.org/property/yearHumidity',
    short_key: 'yearHumidity',
    value: '77',
  }
]
```